backend/app/api/routes/dde.py

```diff
@@ -209,19 +209,32 @@ async def extract_from_file(*, session: SessionDep, current_user: CurrentUser, n
     if not document_data_extractor.owner_id:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="DocumentDataExtractor has no owner")
     # Build examples
-    examples = tup(*(tup(as_text(document_data_extractor.owner, example.document_id), example.data) for example in document_data_extractor.document_data_examples))
+    examples = tup(*(tup(as_text(document_data_extractor.owner, example.document_id, example.start_page, example.end_page), example.data) for example in document_data_extractor.document_data_examples))
     # Pull data from the file
     if upload.content_type != 'application/pdf':
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="This endpoint can only process pdfs")
     prompt = pdf_reader.as_text(upload.file.read())
-    # TODO Marta can improve this
-    chat_completion_request = ChatCompletionRequest(
-        model='gpt-3.5-turbo',
-        messages=[ChatCompletionMessage(role="system", content=document_data_extractor.prompt)]+
-            [ChatCompletionMessage(role=role, content=content) for example in await examples.evaluate() for role, content in [("user", example[0]), ("assistant", example[1])]]+
-            [ChatCompletionMessage(role="user", content=prompt)]
-    ).model_dump(exclude_unset=True)
+    system_prompt = document_data_extractor.prompt
+    examples_text = ""
+    for example in await examples.evaluate():
```

Comment:
you know you can type:
`for input_text, output_text in await examples.evaluate():
  ...`